corehq/apps/smsbillables/models.py

```diff
@@ -273,23 +284,29 @@ class Meta:
 
     @property
     def gateway_charge(self):
-        return self.multipart_count * self._single_gateway_charge
+        if self.direct_gateway_fee is None:
+            return self.multipart_count * self._single_gateway_charge
+        else:
+            return self._single_gateway_charge
 
     @property
     def usage_charge(self):
         return self.multipart_count * self._single_usage_charge
 
     @property
     def _single_gateway_charge(self):
+        amount = None
         if self.gateway_fee is not None:
             try:
-                charge = SmsGatewayFee.objects.get(id=self.gateway_fee.id)
-                if self.gateway_fee_conversion_rate is not None:
-                    return charge.amount / self.gateway_fee_conversion_rate
-                return charge.amount
-            except ObjectDoesNotExist:
+                amount = SmsGatewayFee.objects.get(id=self.gateway_fee.id).amount
```

Comment:
i think this it's worth fixing this line even if it was already like this - it's going to incur an extra db lookup each time we process this and i'm not sure why it's needed?

also i think we can further optimize by setting `gateway_fee = self.gateway_fee` and then using `gateway_fee` throughout this method